File: app/src/main/java/yagoc/Player.java
package yagoc;

public interface Player {
    Move move(Board board);

    PlayerType type();

    PieceColor pieceColor();

    String name();

    default boolean isUser() {
        return type() == PlayerType.user;
    }

    default boolean isComputer() {
        return type() == PlayerType.computer;
    }
}

File: yagoc-desktop/src/main/kotlin/jenm/yagoc/players/Player.kt
package jenm.yagoc.players

import jenm.yagoc.board.BoardView
import jenm.yagoc.board.Move
import jenm.yagoc.pieces.PieceColor

abstract class Player(val name: String, val pieceColor: PieceColor, val type: PlayerType) {
    abstract fun move(board: BoardView): Move
    val isUser = type == PlayerType.User
    val isComputer = type == PlayerType.Computer
}